source/simulator/checkpoint_restart.cc

```diff
@@ -181,49 +181,42 @@ namespace aspect
   }
 
   template <int dim>
-  void Simulator<dim>::create_snapshot()
+  void Simulator<dim>::create_snapshot(bool rotating_checkpoint)
```

Comment:
Added a parameter that controls rotating checkpoints versus permanent checkpoint files!